src/storage_tiering.cpp

```diff
@@ -757,8 +757,24 @@ namespace irods {
          };
 
         execMyRuleInp_t exec_inp{};
-        rstrcpy(exec_inp.myRule, rule_obj.dump().c_str(), META_STR_LEN);
         msParamArray_t* out_arr{};
+        const auto free_inputs_and_outputs = irods::at_scope_exit{[&exec_inp, out_arr] {
```

Comment:
`out_arr` needs to be captured by reference.

Reply:
Is that because the pointed-to struct is being modified? I have been told in the past to capture pointers by value. Just would like to understand why.

Reply:
The pointer is null when it's captured. Therefore, when the lambda is executed, it will have a nullptr value. The pointer must be pointing to the target object before the capture or be taken by reference for the lambda to see the pointed to object.

Reply:
Ah, duh. That makes sense. Also means this isn't what was leaking here. Will update

Reply:
Done